services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -1193,6 +1576,11 @@ protected void initP2PNetwork() {
         new LocalOperationAcceptedFilter<>(p2pNetwork::publishVoluntaryExit));
     blsToExecutionChangePool.subscribeOperationAdded(
         new LocalOperationAcceptedFilter<>(p2pNetwork::publishSignedBlsToExecutionChange));
+
+    this.nodeId =
+        p2pNetwork
+            .getDiscoveryNodeId()
+            .orElseThrow(() -> new InvalidConfigurationException("NodeID is required"));
```

Comment:
Maybe something more friendly to user, say "Failed to get NodeId from discovery system"